src/transformers/models/gpt_sw3/tokenization_gpt_sw3.py

```diff
@@ -0,0 +1,267 @@
+import os
+import re
+import unicodedata
+
+from ... import is_torch_available
+
+
+if is_torch_available():
+    import torch
+
+from shutil import copyfile
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import sentencepiece as spm
+
+from ...tokenization_utils import PreTrainedTokenizer
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+VOCAB_FILES_NAMES = {"vocab_file": "spiece.model"}
+
+# This does not seem to be used at all except for some unit testing? The file is found with the above:
+#  VOCAB_FILES_NAMES, which is sufficient to find the tokenizer since there is one per model size
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "AI-Sweden/gpt-sw3-126m": "https://huggingface.co/AI-Sweden/gpt-sw3-126m/resolve/main/spiece.model",
+        "AI-Sweden/gpt-sw3-350m": "https://huggingface.co/AI-Sweden/gpt-sw3-350m/resolve/main/spiece.model",
+        "AI-Sweden/gpt-sw3-1.6b": "https://huggingface.co/AI-Sweden/gpt-sw3-1.6b/resolve/main/spiece.model",
+        "AI-Sweden/gpt-sw3-6.7b": "https://huggingface.co/AI-Sweden/gpt-sw3-6.7b/resolve/main/spiece.model",
+        "AI-Sweden/gpt-sw3-20b": "https://huggingface.co/AI-Sweden/gpt-sw3-20b/resolve/main/spiece.model",
+    }
+}
+
+# This does not seem to be used except for prompting a warning when tokenizing sequences longer than these
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "AI-Sweden/gpt-sw3-126m": 2048,
+    "AI-Sweden/gpt-sw3-350m": 2048,
+    "AI-Sweden/gpt-sw3-1.6b": 2048,
+    "AI-Sweden/gpt-sw3-6.7b": 2048,
+    "AI-Sweden/gpt-sw3-20b": 2048,
+}
+
+
+class GPTSw3Tokenizer(PreTrainedTokenizer):
```

Comment:
Same comment for function that are the same as `GPT2` let's add the `# Copied from` statements